src/doom/g_game.c

```diff
@@ -972,7 +972,31 @@ static void SetMouseButtons(unsigned int buttons_mask)
 // Get info needed to make ticcmd_ts for the players.
 // 
 boolean G_Responder (event_t* ev) 
-{ 
+{
+    // [crispy] demo fast-forward
+    if (ev->type == ev_keydown && ev->data1 == key_demospeed && 
+        (demoplayback || gamestate == GS_DEMOSCREEN))
+    {
+        if (singletics == false)
+        singletics = true;
+        else singletics = false;
+        return true;
+    }
+
+    // [crispy] demo pause (from prboom-plus)
+    if (gameaction == ga_nothing && 
+        (demoplayback || gamestate == GS_DEMOSCREEN))
+    {
+         if (ev->type == ev_keydown && ev->data1 == key_pause)
+    {
```

Comment:
Indentation is somehow borked up here.

Reply:
I couldn't figure out the correct way when importing from prb+. It doesn't seem to be consistent elsewhere in this file. Another attempt at getting it right in 9430f1e. If that still doesn't look right I'll need explicit instructions on how it should be done.